core/runtime/src/main/java/io/quarkus/runtime/RuntimeValuesConfigSource.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.runtime;
+
+import java.util.Set;
+
+import io.smallrye.config.common.AbstractConfigSource;
+
+public class RuntimeValuesConfigSource extends AbstractConfigSource {
+
+    public RuntimeValuesConfigSource() {
+        super("io.quarkus.runtime.RuntimeValuesConfigSource", Integer.MAX_VALUE - 10);
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return Set.of();
+    }
+
+    @Override
+    public String getValue(String propertyName) {
+        // TODO - avoid alloc for lookup
+        Object value = RuntimeValues.config().get(propertyName);
+        return value != null ? value.toString() : null;
+    }
+}
```

Comment:
I think this compatibility layer is completely unnecessary, and probably harmful. I would just introduce the new mechanism, by itself, and then slowly start moving things over to use it.

Changing the code to register things into the map does not break compatibility in any way. Then, changing code to read from the map also doesn't break anything.

Reply:
OK. If you feel strongly about it, then I'm with you. But we do need to ensure the thread-safety of what's going on here.